src/Events/EventContextDataContainer.php

```diff
@@ -0,0 +1,136 @@
+<?php
+declare(strict_types=1);
+
+namespace LotGD\Core\Events;
+
+use LotGD\Core\Exceptions\ArgumentException;
+
+/**
+ * Abstract EventContextDataContainer to provide a basic structure for managing contextual data of an event.
+ *
+ * This class must be immutable and returns always a new instance of itself for any change.
+ * @package LotGD\Core\Events
+ * @immutable
+ */
+abstract class EventContextDataContainer
```

Comment:
Why make this abstract and then make a completely empty subclass called `EventContextData`? I would just make this a regular class called `EventContextData`.